packages/eslint-plugin/docs/rules/prefer-readonly-parameter-types.md

```diff
@@ -126,18 +126,103 @@ interface Foo {
 ## Options
 
 ```ts
+type TypeAllowlistItem = {
+  typeName: string;
+} & ({ local: true } | { defaultLib: true } | { package: string });
```

Comment:
Putting a little more thought into this: I'm now wondering if it would make more sense to use a traditional [discriminated union](https://basarat.gitbook.io/typescript/type-system/discriminated-unions)...

```ts
} & ({ source: 'default-lib' } | { source: 'local' } | { package: string; source: 'package' });
```

It's a little easier to check in code against the one property. One discriminant type is easier to remember than three boolean/string properties, too...

Thoughts?

cc @bradzacher too as this might end up being a type used across other options.

Reply:
Yeah, that's quite clearly a better solution.